webui/react/src/pages/F_ExpList/CompareMetrics.tsx

```diff
@@ -59,14 +61,18 @@ const CompareMetrics: React.FC<Props> = ({ selectedExperiments, trials }) => {
 
   return (
     <div style={{ height: 'calc(100vh - 250px)', overflow: 'auto' }}>
-      <ChartGrid
-        chartsProps={chartsProps}
-        handleError={handleError}
-        scale={scale}
-        setScale={setScale}
-        xAxis={xAxis}
-        onXAxisChange={setXAxis}
-      />
+      {isLoaded ? (
+        <ChartGrid
+          chartsProps={chartsProps}
+          handleError={handleError}
+          scale={scale}
+          setScale={setScale}
+          xAxis={xAxis}
+          onXAxisChange={setXAxis}
+        />
+      ) : (
+        <Spinner center spinning />
+      )}
```

Comment:
nit: consider instead:
```suggestion
      <Spinner center spinning={isLoaded}>
        <ChartGrid
          chartsProps={chartsProps}
          handleError={handleError}
          scale={scale}
          setScale={setScale}
          xAxis={xAxis}
          onXAxisChange={setXAxis}
        />
      </Spinner>
```

actually -- how do you feel about using the ternary versus using the `spinning` prop to control display? I think that ideally the `Spinner` component would only support one or the other, maybe only the ternary because it can act as a type guard. Having both leads to errors like the issue you fixed in [CompareParallelCoordinates.tsx](https://github.com/determined-ai/determined/pull/7370/files#diff-7484fab5839d77641ee6b0c9815348d1071688725de05755e658f24a7542668f).

Reply:
While you're talking abstractions, I'd also like to propose:
```typescript
interface SpinnerProps<T> {
  center: boolean,
  data: Loadable<T>,
  children: (data: T) => ReactNode
}
...
<Spinner center data={chartProps}>{chartProps => (
  <ChartGrid
    chartsProps={chartsProps}
    handleError={handleError}
    scale={scale}
    setScale={setScale}
    xAxis={xAxis}
    onXAxisChange={setXAxis}
  />
)}</Spinner>
```